dot/parachain/statement-distribution/statement_distribution.go

```diff
@@ -2,69 +2,84 @@ package statementdistribution
 
 import (
 	"context"
+	"time"
 
 	"github.com/ChainSafe/gossamer/internal/log"
 
 	statementedistributionmessages "github.com/ChainSafe/gossamer/dot/parachain/statement-distribution/messages"
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	parachainutil "github.com/ChainSafe/gossamer/dot/parachain/util"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "statement-distribution"))
 
 type StatementDistribution struct {
+	reputationAggregator *parachainutil.ReputationAggregator
+	SubSystemToOverseer  chan<- parachainutil.NetworkBridgeTxMessage
 }
 
-func (s StatementDistribution) Run(ctx context.Context, overseerToSubSystem <-chan any) {
+func (s *StatementDistribution) Run(
+	ctx context.Context,
+	overseerToSubSystem <-chan any,
+	v2CommChannel <-chan any,
+	receiverRespCh <-chan any,
+	retryReqCh <-chan any,
```

Comment:
In [this comment](https://github.com/ChainSafe/gossamer/pull/4406#issuecomment-2545580713) @EclesioMeloJunior said that these channels are created in the `Run()` method. However, having just read the issue I'm not really sure how it is supposed to work exactly. Therefore I'll defer to him on this one. :)

Reply:
I only realized now that those "other channels" are related to request/response network protocols. While designing the availability subsystem I noticed that we don't have a way yet to use those in parachain subsystems.

I created #4448 to address this, but with the way this issue is currently written, it only covers sending out requests from a subsystem and receiving the response. Whereas here we need to listen to requests from other peers (which the availability subsystem needs to do as well, so #4448 is also insufficient for that, but that's off-topic here :).

So the two conclusions from these realizations are:
1. Adding request/response support to network bridge is a bit more involved than I thought.
2. Without having that, I'm not sure how we should proceed here.

Reply:
@haikoschol this PR is to introduce the fan in skeleton to the statement distribution subsystem, for this PR the request/response channel is not a priority given that it will be added latter.